rasa/shared/core/domain.py

```diff
@@ -1197,14 +1198,9 @@ def is_domain_file(filename: Text) -> bool:
 
         if not is_likely_yaml_file(filename):
             return False
-        try:
```

Comment:
this is a boolean check. Isn't it supposed to not throw things? The least we should do is document the exception in the docstring as it's not intuitive.

Reply:
yes I agree, it is not ideal. but just returning `False` will mean invalid files just get ignored, which isn't really helpful either. ideally we'd split test if a file is valid yaml before calling `is_domain` / `is_story` / ...

Reply:
or we make all the boolean checks independent of yaml reading (test against the text directly)?

Reply:
feel free to move forward but please create an issue to find a more robust solution after the release.